### ISSUE TITLE
Website count broken for extensions no longer installed on a wiki

### ISSUE THREAD

**Issue by kghbln** [WikiApiary/WikiApiary#180]
See this [extension](https://wikiapiary.com/wiki/Extension:Semantic_Text_Annotator) @Cavila added today

**Comment by plexish** [comment]
Worse than that, website count is broken for all extensions. https://wikiapiary.com/wiki/Extension:ConfirmEdit https://wikiapiary.com/wiki/Extension:ParserFunctions

And it was working earlier today, with no edits which look likely to be able to cause that. I'm guessing there's an issue with the semantic rating update, or possibly some other recent extension change.

**Comment by kghbln** [comment]
Oops, have not had a look at these so I did not realise that the issue is wider spread than expected. But why should the update of Semantic Rating break the website count or is this woven together right now?

**Comment by jthingelstad** [comment]
Read the error message.

```
Subqueries not supported for values of property "Has extension".
The part "Extension:ParserFunctions" of the query was not understood. Results might not be as expected.
The part "
"&lt;spanclass="smw-highlighter"data-type="4"data-state="inline"data-title="Error"&gt;&lt;spanclass="smwtticonwarning"&gt;&lt;/span&gt;&lt;spanclass="smwttcontent"&gt;&lt;ul&gt;&lt;li&gt;Subqueriesnotsupportedforvaluesofproperty"Hasextension".&lt;/li&gt;&lt;li&gt;Thepart"Extension:ParserFunctions"ofthequerywasnotunderstood.
Resultsmightnotbeasexpected.&lt;/li&gt;&lt;li&gt;Thepart"" is not declared as a valid unit of measurement for this property." of the query was not understood.
Results might not be as expected.
```

I think this is due to the Semantic MediaWiki update to 2.0-rc3.

**Comment by jthingelstad** [comment]
I confirmed the issue is with the `ask` query. If you see [this test page](https://wikiapiary.com/wiki/User:Thingles/foo) where I just have the one query you can see it returns the valid result but it also gives the warning.

`{{#ask:[[Has subobject.Has extension::Extension:ConfirmEdit]] [[Is in farm::true]] [[Is defunct::false]] | format=count }}
`

The warning then cascades into a bunch of other errors on Template:Extension.

@kghbln @etesp I’m leaving town so can’t look further for now. This is almost definitely triggered by #177 and the update to Semantic MediaWiki 2.0-rc3. I think the query just needs to be fixed or changed to fix this. I guess it’s equally as likely this is a bug in Semantic MediaWiki since it worked fine before. I don't have time to submit now but maybe one of you could?

**Comment by plexish** [comment]
I can try and fix but I'm not going to have much time over the next couple of days, then I'm away from the net entirely for a week.

**Comment by jthingelstad** [comment]
I also created SemanticMediaWiki/SemanticMediaWiki#452. Something changed here, but it's not clear if it changed to be correct or not.

**Comment by jthingelstad** [comment]
@mwjames picked up this issue and it does appear to be a bug. Disregard any attempt to fix. Well just get updated code soon.

**Comment by mwjames** [comment]
> just get updated code soon

I need to write some proper tests first since this was the issue in the first place so we don't fall short next time.

**Comment by jthingelstad** [comment]
Happy I finally got the update out in so WikiApiary could help test and find stuff before release! Thanks @mwjames! I’m eager to see if the properties issue is gone. Should know by tonight.

**Comment by mwjames** [comment]
> I’m eager to see if the properties issue is gone. Should know by tonight.

Well, it is not gone but we hide the fact that the `RefreshLinksJob` does something it shouldn't do therefore you will find `Empty SemanticData / re-parsing: ...` messages in your normal log which means for those items you got an empty container and extra work (namely page parsing) has to be done in order to retrieve the lost information from the `ParserOutput`.

**Comment by jthingelstad** [comment]
Closed with #182.
